### DISCUSSION TITLE
Not fully understanding Auto DROP

### DISCUSSION THREAD

**Discussion by bowerhaus** [artisan-roaster-scope/artisan#1159] [Q&A] [answered]
Hi,

I am using Artisan 2.8.2 running on a Raspberry Pi Model B to automate via Phidgets a home build roaster based on the "flour sifter" designs that are kicking around. I have a 3mm dia RTD to monitor bean temperature (no ET monitoring) and an SSR using PWM to control the heater (the innards out of a hot air gun).

I'm using the PID automation to follow a background profile but I'm having trouble understanding how the Auto DROP is meant to work. I've attached an image of the last roast and you can see that the BT tracked the curve pretty well. The DE point was a little early but I get that, since it is based on the temperature at the top of the drying phase (taken from the background DE point). Similarly, FC looks good and, again, taken from the temperature of the end of the maillard phase. However, the DROP was late but I don't understand why this point was chosen. The temperature 208.4 was not the same as the background profile drop (206.5) and the time was not the same either.

Can someone explain how the Auto DROP is calculated and what the best way to get it to match the temp/time on a background profile is. Ideally, I'd like the DROP to be at the same development percentage taken from the profile.

I'd be grateful for any pointers, thanks

![LatestRoast](https://user-images.githubusercontent.com/2306120/236833858-e64bbdc9-d972-4487-8247-18a4bc0c1d6e.png)
[TestProfileAlog.txt](https://github.com/artisan-roaster-scope/artisan/files/11421427/TestProfileAlog.txt)
[bh2-artisan-settings-aset.txt](https://github.com/artisan-roaster-scope/artisan/files/11421431/bh2-artisan-settings-aset.txt)
![Roaster](https://user-images.githubusercontent.com/2306120/236834569-11d9866e-edf0-4bec-b19c-a4d0ba6452da.jpg)

**Comment by MAKOMO** [comment, marked as answer]
The AutoDROP mechanism configurable under menu Config >> Events, 1st tab, has nothing to do with automation. This is a heuristic that tries to recognise the sudden temperature drop at the end of a roast on DROP to place the DROP mark in Artisan automatically such that the operator does not has to click the DROP button at that moment. This works well on most drum roasters, but your profiles seem not to have a clear enough temperature drop to have this mechanism to work correctly. Anyhow, this mechanism does not take any information from the background profile into account.

In the Background dialog (menu Roast >> Background) there is an option named "Playback DROP" which can set the DROP mark based on the background profile either by time or temperature. This is most useful in connection with an automated door mechanism which can be triggered by the DROP button to automatically release the beans at that moment.

**Reply by bowerhaus** [reply]
That would explain it - thanks

**Comment by Terracotta-6** [comment]
Artisan Continuous build 56b381d
Verify Playback DROP results
Playback DROP is triggered based on background profile time or temperature, Priority of triggering DROP condition. Temperature takes precedence.
![9bcaeac8bd032e782aab86252b3d94a](https://github.com/user-attachments/assets/fc2cc5e2-bd57-4c20-a95d-0cde2b9993d2)
![e397c6cab4691139911e1038fc39090](https://github.com/user-attachments/assets/2e214522-6f56-47fb-ace7-2e60153d2018)
